tasks/shell.sh

```diff
@@ -0,0 +1,555 @@
+      fi
+      rpm -Uvh --oldpackage --replacepkgs "$2"
+      if test "$version" = 'latest'; then
+        yum install -y puppet-agent
+      else
+        yum install -y "puppet-agent-${puppet_agent_version}"
+      fi
+      ;;
+    "deb")
+      info "installing puppetlabs apt repo with dpkg..."
+      dpkg -i "$2"
+      apt-get update -y
+      if test "$version" = 'latest'; then
+        apt-get install -y puppet-agent
+      else
+        if test "x$deb_codename" != "x"; then
+          apt-get install -y "puppet-agent=${puppet_agent_version}-1${deb_codename}"
+        else
+          apt-get install -y "puppet-agent=${puppet_agent_version}"
+        fi
+      fi
+      ;;
+    "solaris")
+      critical "Solaris not supported yet"
+      ;;
+    "dmg" )
+      critical "Puppet-Agent Not Supported Yet: $1"
+      ;;
+    *)
+      critical "Unknown filetype: $1"
+      report_bug
+      exit 1
+      ;;
+  esac
+  if test $? -ne 0; then
+    critical "Installation failed"
+    report_bug
+    exit 1
+  fi
+}
+
+#Platforms that do not need downloads are in *, the rest get their own entry.
+case $platform in
+  "archlinux")
+    critical "Not got Puppet-agent not supported on Arch yet"
+    ;;
+  "freebsd")
+    critical "Not got Puppet-agent not supported on freebsd yet"
+    ;;
+  *)
+    info "Downloading Puppet $version for ${platform}..."
+    case $platform in
+      "el")
+        info "Red hat like platform! Lets get you an RPM..."
+        filetype="rpm"
+        filename="puppet5-release-el-${platform_version}.noarch.rpm"
+        download_url="http://yum.puppetlabs.com/puppet5/${filename}"
+        ;;
+      "fedora")
+        info "Fedora platform! Lets get the RPM..."
+        filetype="rpm"
+        filename="puppet5-release-fedora-${platform_version}.noarch.rpm"
+        download_url="http://yum.puppetlabs.com/puppet5/${filename}"
+        ;;
+      "debian")
+        info "Debian platform! Lets get you a DEB..."
+        case $major_version in
+          "5") deb_codename="lenny";;
+          "6") deb_codename="squeeze";;
+          "7") deb_codename="wheezy";;
+          "8") deb_codename="jessie";;
+          "9") deb_codename="stretch";;
+        esac
+        filetype="deb"
+        filename="puppet5-release-${deb_codename}.deb"
+        download_url="http://apt.puppetlabs.com/${filename}"
+        ;;
+      "ubuntu")
+        info "Ubuntu platform! Lets get you a DEB..."
+        case $platform_version in
+          "12.04") deb_codename="precise";;
+          "12.10") deb_codename="quantal";;
+          "13.04") deb_codename="raring";;
+          "13.10") deb_codename="saucy";;
+          "14.04") deb_codename="trusty";;
+          "15.04") deb_codename="vivid";;
+          "15.10") deb_codename="wily";;
+          "16.04") deb_codename="xenial";;
+          "16.10") deb_codename="yakkety";;
+          "17.04") deb_codename="zesty";;
```

Comment:
I have been using this for testing `apply` on ubuntu 18.04 by adding `"18.04") deb_codename="bionic";;`

Reply:
I'll go ahead and add that